aesara/tensor/shape.py

```diff
@@ -885,3 +891,142 @@ def shape_padaxis(t, axis):
+        The dimension along which the tensor x should be broadcastable.
+        If the length of x along these dimensions is not 1, a ValueError will
+        be raised.
+
+    Returns
+    -------
+    tensor
+        A aesara tensor, which is broadcastable along the specified dimensions.
+
+    """
+    x = as_tensor_variable(x)
+
+    if not axes:
+        return x
+
+    if max(axes) >= x.type.ndim:
+        raise ValueError("Trying to specify broadcastable of non-existent dimension")
+
+    shape_info = [1 if i in axes else None for i in range(len(x.type.shape))]
+    return specify_shape(x, shape_info)
+
+
+class MaskBroadcastable(COp):
+    """
+    Mask static broadcastable dimensions of input as `None`
+
+    See Also
+    --------
+    mask_broadcastable <aesara.tensor.shape.mask_broadcastable>
+
+
+    Examples
+    --------
+    ``MaskBroadcastable((1,))(x)`` would make `x` second static dimension be `None`
+
+    """
+
+    view_map = {0: [0]}
+    _f16_ok = True
+    # Mapping from Type to C code (and version) to use.
+    # In the C code, the name of the input variable is %(iname)s,
+    # the output variable is %(oname)s.
+    c_code_and_version: Dict = {}
+
+    check_input = False
+    __props__ = ("axis",)
+    _f16_ok = True
+
+    def __init__(self, *axis):
+        # Sort them to make sure we merge all possible case.
+        items = tuple(sorted(axis))
+        self.axis = items
+        for axis in self.axis:
+            if not isinstance(axis, (np.integer, int)):
+                raise TypeError(f"MaskBroadcastable needs integer axes. Got {axis}")
+
+    def __str__(self):
+        return f"{self.__class__.__name__}{{{','.join(str(i) for i in self.axis)}}}"
+
+    def make_node(self, x):
+        x = as_tensor_variable(x)
+        if x.type.ndim <= max(self.axis):
+            raise ValueError("Trying to mask broadcastable of non-existent dimension")
+        shape = list(x.type.shape)
+        for axis in self.axis:
+            shape[axis] = None
+        return Apply(self, [x], [x.type.clone(shape=shape)()])
+
+    def perform(self, node, inp, out_):
+        (x,) = inp
+        (out,) = out_
+        out[0] = x
+
+    def grad(self, inp, grads):
+        (x,) = inp
+        (gz,) = grads
+        # restore the broadcasting pattern of the input
+        return [specify_shape(gz, x.type.shape)]
+
+    def infer_shape(self, fgraph, node, ishapes):
+        assert len(ishapes) == 1
+        return [tuple(ishapes[0])]
+
+    def R_op(self, inputs, eval_points):
+        if eval_points[0] is None:
+            return [None]
+        return self(*eval_points, return_list=True)
+
+    def c_code(self, node, nodename, inp, out, sub):
+        (iname,) = inp
+        (oname,) = out
+
+        return f"""
+        Py_XDECREF({oname});
+        {oname} = {iname};
+        Py_XINCREF({oname});
+        """
+
+    def c_code_cache_version(self):
+        return (3,)
+
+
+def mask_broadcastable(x, *axes):
```

Comment:
Again, `remove_shape_info` or something similar would be much more accurate.

Reply:
I called ot that because it's only ever used to remove info of `shape==1`